### PR TITLE
output a warning on publish

### PR DESCRIPTION
**Pull request by jrwren**
output a warning on publishing to stable channel if bugs-url and homepage are not set

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/57/
Test PASSed.

File: cmd/charm/charmcmd/publish.go [hunk 1]
```diff
@@ -116,6 +116,31 @@ func (c *publishCommand) Run(ctxt *cmd.Context) error {
 	}
 	fmt.Fprintln(ctxt.Stdout, "url:", c.id)
 	fmt.Fprintln(ctxt.Stdout, "channel:", c.channel)
+	if "stable" == c.channel {
```
Comment (makyo) [inline cmd/charm/charmcmd/publish.go:119]: Is it Go-esque to compare in this order?

**Comment by makyo** [comment]
:+1:

File: cmd/charm/charmcmd/publish.go (hunk 1 above)
Comment (cmars) [inline cmd/charm/charmcmd/publish.go:119]: No. That defensive style makes sense coming from a C background, which allows assignments in an if statement if you forgot an `=`. It's neither necessary nor idiomatic to do this in Go. The compiler will prevent unintentional assignment in an if expression.

File: cmd/charm/charmcmd/publish.go [hunk 2]
```diff
@@ -116,6 +116,31 @@ func (c *publishCommand) Run(ctxt *cmd.Context) error {
...
+		} else {
+			if v, ok := commonInfo["bugs-url"].(string); !ok || v == "" {
+				unset = "bugs-url"
+			}
+			if v, ok := commonInfo["homepage"].(string); !ok || v == "" {
+				if unset != "" {
+					unset += " and "
+					verb = "are"
+				}
+				unset += "homepage"
```
Comment (bac) [inline cmd/charm/charmcmd/publish.go:137]: The += is confusing since you just determined `unset` is null.  Just do an assignment.

**Comment by bac** [comment]
:+1:, no-qa

File: cmd/charm/charmcmd/publish.go (hunk 1 above)
Comment (jrwren) [inline cmd/charm/charmcmd/publish.go:119]: Old habits die hard.

**Comment by jrwren** [comment]
QA instructions:
1. go install ./cmd/charm/charmcmd/
2. cd $GOPATH/src/gopkg.in/juju/charm.v6-unstable/internal/test-charm-repo/bundle
3. export JUJU_CHARMSTORE=http://api.jujugui.org/charmstore
4. charm push wordpress-simple wordpress-simple
5. charm publish cs:~evarlast/bundle/wordpress-simple-1 (or whatever url was returned from previous step)
6. See the warning in output from this last command.
7. charm set (whatever url was used in last command) bugs-url=http://idunno.example.com homepage=http://istilldunno.example.com
8. charm publish cs:~evarlast/bundle/wordpress-simple-1 (or whatever url was returned from previous step)
9. See there is no warning this time.

**Comment by mhilton** [comment]
QA :ok: 

```
$ charm push test-bundle wordpress-simple
url: cs:~martin-hilton/bundle/wordpress-simple-0
channel: unpublished
$ charm publish cs:~martin-hilton/bundle/wordpress-simple-0
url: cs:~martin-hilton/bundle/wordpress-simple-0
channel: stable
warning: bugs-url and homepage are not set.  See set command.
$ charm set cs:~martin-hilton/bundle/wordpress-simple-0 bugs-url=http://idunno.example.com homepage=http://istilldunno.example.com
$ charm publish cs:~martin-hilton/bundle/wordpress-simple-0
url: cs:~martin-hilton/bundle/wordpress-simple-0
channel: stable
```

**Comment by jrwren** [comment]
:shipit: 

Thanks @mhilton sorry about the bad bundle suggestion.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**